pages/contribution-flow.js

```diff
@@ -106,10 +108,18 @@ class NewContributionFlowPage extends React.Component {
     loadingLoggedInUser: PropTypes.bool,
     tags: PropTypes.arrayOf(PropTypes.string),
     step: PropTypes.oneOf(Object.values(STEPS)),
+    router: PropTypes.object,
   };
 
   componentDidMount() {
     this.loadExternalScripts();
+    const { router, data } = this.props;
+    const account = data?.account;
+    let path = router.asPath;
+    if (!path.includes(account?.parent?.slug)) {
+      path = path.replace(`${account?.slug}/`, '');
+      addParentToURLIfMissing(router, account, path);
```

Comment:
Parent slug: `babel`
Child slug: `babel-project`
Result: No redirect

Let's not try to re-parse the URL, we have the `router.query` object for this.

Reply:
@Betree : Thanks much for the reviews. I've simplified this part as I realized the `if` statement doesn't need to be there at all and now only used the `addParentToURLIfMissing`. I think this should take care of this issue. 🤔